### ISSUE TITLE
Function.prototyp.toString only returns a fixed `[native code]` string

### ISSUE THREAD

**Issue by FredrikNoren** [boa-dev/boa#3780]
**ECMASCript feature**
Function.prototyp.toString returns the actual javascript implementation in v8. [ECMAScript specification][spec]

[spec]: https://tc39.es/ecma262/#sec-function.prototype.tostring

**Example code**

```javascript
let f = () => 1 + 1;
console.log(f.toString());
```
The expected output is `() => 1 + 1`.

**Comment by Nikita-str** [comment]
Is it ok to add in [Cursor](https://github.com/boa-dev/boa/blob/main/core/parser/src/parser/cursor/mod.rs#L25) a fields that will collect `source_text: Vec<u32>` when callable is parsing (if nested callable is parsing then it just will take slice of it) to solve the issue?
Or maybe do it at some lower layer(on `BufferedLexer` or `Lexer`)?
Or is this a bad approach?
I don't find any better alternative

**Comment by jedel1043** [comment]
Yeah, those changes sound good. I would say that on the AST side it would be the Module and the Script the ones holding the script source, and the rest of nodes would just have a span to index that.

**Comment by Nikita-str** [comment]
Hmm, do I understand you correctly: collect ALL the source code, and use it whenever we need some part of initial code (e.g. name of functions, string literals, and function source codes)?

Or should it be done only for source codes of functions (it's what I try to do now)?

**Comment by jedel1043** [comment]
All the code probably, because we're planning to use that to improve the error messages.
